gotrue/_async/__init__.py

```diff
@@ -0,0 +1 @@
+from __future__ import annotations
```

Comment:
Why is this needed in the init file?

Reply:
Because if code is added to that file in the future, it is less likely that we will forget to include that `import` for the precommit `pyupgrade` hooks to act and force the appropriate changes to be made. Because it could happen that when adding code we forget to include the `import` and `pyupgrade` will not force the changes.

PS: `pyupgrade` requires the `import` to be done

What do you think about that?